plugin/src/main/java/com/denizenscript/denizen/utilities/TextWidthHelper.java

```diff
@@ -3,30 +3,32 @@
 import com.denizenscript.denizencore.utilities.AsciiMatcher;
 import org.bukkit.ChatColor;
 
+import java.util.HashMap;
+
 public class TextWidthHelper {
 
-    public static int[] characterWidthMap = new int[128];
+    public static HashMap<Character, Integer> characterWidthMap = new HashMap<>();

```

Comment:
This particular change represents a very significant performance degradation for large strings

I imagine the same unicode-compat could be achieved with just a slightly larger array - or if not, then an array for low values and a map for high values

Reply:
Some of these have values over 9000 (insert meme)
```
for (char c : "▒▓└┴┬".toCharArray()) {
    System.out.println((int) c);
}

9618
9619
9492
9524
9516
```
I'll commit a more performant change, though I can't realistically expect people to use text_width on long strings often